Sources/BraintreeCoreSwift/BTHTTP.swift

```diff
@@ -0,0 +1,542 @@
+            guard let request = request else {
+                self.handleRequestCompletion(data: nil, request: nil, shouldCache: false, response: nil, error: error, completion: completion)
+                return
+            }
+
+            let task: URLSessionTask = self.session.dataTask(with: request) { [weak self] data, response, error in
+                self?.handleRequestCompletion(data: data, request: request, shouldCache: false, response: response, error: error, completion: completion)
+            }
+
+            task.resume()
+        }
+    }
+
+    func createRequest(
+        method: String,
+        path: String?,
+        parameters: NSDictionary? = [:],
+        completion: @escaping (URLRequest?, Error?) -> Void
+    ) {
+        let hasHTTPPrefix: Bool = path?.hasPrefix("http") ?? false
+        let baseURLString: String = baseURL.absoluteString
+        var errorUserInfo: [String: Any] = [:]
+
+        if hasHTTPPrefix && (baseURLString.isEmpty || baseURLString == "") {
+            errorUserInfo["method"] = method
+            errorUserInfo["path"] = path
+            errorUserInfo["parameters"] = parameters
+
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
+
+            completion(nil, error)
+            return
+        }
+        
+        let fullPathURL: URL?
+        let isNotDataURL: Bool = baseURL.scheme != "data"
+
+        if isNotDataURL, let path = path {
+            fullPathURL = hasHTTPPrefix ? URL(string: path) : baseURL.appendingPathComponent(path)
+        } else {
+            fullPathURL = baseURL
+        }
+
+        let mutableParameters: NSMutableDictionary = NSMutableDictionary(dictionary: parameters ?? [:])
+
+        if authorizationFingerprint != "" {
+            mutableParameters["authorization_fingerprint"] = authorizationFingerprint
+        }
+
+        guard let fullPathURL = fullPathURL else {
+            /// baseURL can be non-nil (e.g. an empty string) and still return nil for -URLByAppendingPathComponent:
+            /// causing a crash when NSURLComponents.componentsWithString is called with nil.
+            errorUserInfo["method"] = method
+            errorUserInfo["path"] = path
+            errorUserInfo["parameters"] = parameters
+            errorUserInfo[NSLocalizedFailureReasonErrorKey] = "fullPathURL was nil"
+
+            let error = Self.constructError(code: .missingBaseURL, userInfo: errorUserInfo)
+
+            completion(nil, error)
+            return
+        }
+
+        buildHTTPRequest(
+            method: method,
+            url: fullPathURL,
+            parameters: mutableParameters,
+            isNotDataURL: isNotDataURL
+        ) { request, error in
+            completion(request, error)
+        }
+    }
+
+    func buildHTTPRequest(
+        method: String,
+        url: URL,
+        parameters: NSMutableDictionary? = [:],
+        isNotDataURL: Bool,
```

Comment:
Nitpick - The negative in the bool naming is a little confusing. `isDataURL` would be easier to read IMO.